### ISSUE TITLE
compilation fail on perlmutter

### ISSUE THREAD

**Issue by aodhanmci** [ECP-WarpX/WarpX#5487]
## Description
I was having problems on Perlmutter and thought a recompilation would help - compilation seems to break at the very end. I am following the WarpX on Perlmutter instructions. I then compiled without the QED tables and it seemed to work ok (I think this was the only thing I changed)

## System information
- Operating system (name and version):
  - [x ] Linux: e.g., Ubuntu 22.04 LTS
  - [ ] macOS: e.g., macOS Monterey 12.4
  - [ ] Windows: e.g., Windows 11 Pro
- Version of WarpX: e.g., latest, 24.10, etc.
- Installation method:
  - [ ] Conda
  - [ ] Spack
  - [ ] PyPI
  - [ ] Brew
  - [x ] From source with CMake
  - [ ] Module system on an HPC cluster
- Other dependencies: yes/no, describe
- Computational resources:
  - [ ] CPU
  - [x ] GPU: e.g., NVIDIA, AMD, etc.

If applicable, please add any additional information about your software environment:
- [ x] CMake: 3.24.3
- [x ] C++ compiler: GNU 12.3.0 with NVCC 12.2.91
- [ ] Python: yes but unsure
- [x ] MPI: e.g., OpenMPI 4.5
- [ ] FFTW: not sure
- [x ] HDF5: e.g., 1.12.0
- [x ] ADIOS2: e.g., 2.8.3
- Other dependencies: yes/no, describe

## Additional information
here is the error message:

/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: lib/libwarpx.2d.MPI.CUDA.DP.PDP.OPMD.FFT.EB.QED.GENQEDTABLES.a(BreitWheelerEngineWrapper.cpp.o): in function `long double boost::math::detail::gamma_imp<long double, boost::math::policies::policy<boost::math::policies::promote_float<false>, boost::math::policies::promote_double<false>, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy>, boost::math::lanczos::lanczos17m64>(long double, boost::math::policies::policy<boost::math::policies::promote_float<false>, boost::math::policies::promote_double<false>, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy> const&, boost::math::lanczos::lanczos17m64 const&) [clone .isra.0]':
tmpxft_0007bbf5_00000000-6_BreitWheelerEngineWrapper.cudafe1.cpp:(.text+0x56c6): undefined reference to `boost::assertion_failed_msg(char const*, char const*, char const*, char const*, long)'
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: lib/libwarpx.2d.MPI.CUDA.DP.PDP.OPMD.FFT.EB.QED.GENQEDTABLES.a(QuantumSyncEngineWrapper.cpp.o): in function `long double boost::math::detail::gamma_imp<long double, boost::math::policies::policy<boost::math::policies::promote_float<false>, boost::math::policies::promote_double<false>, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy>, boost::math::lanczos::lanczos17m64>(long double, boost::math::policies::policy<boost::math::policies::promote_float<false>, boost::math::policies::promote_double<false>, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy, boost::math::policies::default_policy> const&, boost::math::lanczos::lanczos17m64 const&) [clone .isra.0]':
tmpxft_0007bbc1_00000000-6_QuantumSyncEngineWrapper.cudafe1.cpp:(.text+0x5226): undefined reference to `boost::assertion_failed_msg(char const*, char const*, char const*, char const*, long)'
collect2: error: ld returned 1 exit status
gmake[2]: *** [CMakeFiles/app_2d.dir/build.make:110: bin/warpx.2d.MPI.CUDA.DP.PDP.OPMD.FFT.EB.QED.GENQEDTABLES] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:3632: CMakeFiles/app_2d.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2

seems to work when I switch off QED tables though.

thanks,
Aodhan

**Comment by aeriforme** [comment]
Thank you for your question.
We have to update the variable `BOOST_ROOT` in the script `perlmutter_<gpu,cpu>_warpx.profile`. 
It should work with this one for now: 
```
export BOOST_ROOT=/global/common/software/spackecp/perlmutter/e4s-23.08/default/spack/opt/spack/linux-sles15-zen3/gcc-12.3.0/boost-1.83.0-2duyuvmto7nxhdxbmtdzfqta2zd2e6gp
```

It will be fixed with PR #5477

**Comment by aodhanmci** [comment]
perfect all sorted. thank you
